openapscontrib/mmhistorytools/historytools.py

```diff
@@ -114,6 +114,11 @@ def __init__(self, trimmed_history, start_datetime=None, end_datetime=None):
         self.end_datetime = end_datetime
 
         if len(trimmed_history) > 0:
+            if start_datetime is None and end_datetime is not None and duration_hours is not None:
```

Comment:
Would it be worthwhile to add test coverage for the duration argument? You should be able to copy/adapt the existing tests for duration.